### PR TITLE
Channel Page

### PR DESCRIPTION
**Pull request by TroyKomodo**
## Proposed changes

Adding the Channel Page

image

Completes #37 

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/ScuffleTV/scuffle/blob/main/CONTRIBUTING.md) doc
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

Channel page will be where users can see their stream / chat ect

### CODE REVIEW

**Comment by chipsTM** [comment]
Here's a short list of stuff I tested
- new user creation
- logging in and out
- signing up with same user name fails
- channel is accessible from <base_url>/<channel_name>
- not able to chat while not logged in
- able to chat while logged in
- sidebars are collapsible and video expands to fill missing space

Looks good to me

**Review by chipsTM — APPROVED** [review]
Here's a short list of stuff I tested
- new user creation
- logging in and out
- signing up with same user name fails
- channel is accessible from <base_url>/<channel_name>
- not able to chat while not logged in
- able to chat while logged in
- sidebars are collapsible and video expands to fill missing space

Looks good to me

File: frontend/website/src/components/closeSidebar.svelte [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<svg width="1em" height="1em" viewBox="0 0 20 16" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (treuks) [inline frontend/website/src/components/closeSidebar.svelte:1]: replace 1em with 0.7em

File: frontend/website/src/components/topNav.svelte [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+		z-index: 5;
+	}
+
+	.search {
+		display: flex;
+		width: 30rem;
+		position: relative;
+
+		& > input {
+			border: 3px solid $borderColor;
```
Comment (treuks) [inline frontend/website/src/components/topNav.svelte:111]: although 3px looks kinda cool, to make it fit in more with the style of the website it should be 1px

File: frontend/website/src/components/topNav.svelte [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+		transition: background-color 0.5s, color 0.5s, box-shadow 0.5s;
+		cursor: pointer;
+		padding: 0.5rem 1rem;
+		margin: 0 0.5rem;
+		font: inherit;
+		border: 1px solid #96491c;
+		background: #16181862;
+		box-shadow: 0px 6px 20px 7px rgba(255, 115, 87, 0.1);
+		&.signup {
+			background-color: #cf634d;
```
Comment (treuks) [inline frontend/website/src/components/topNav.svelte:163]: the color is a bit too dark and looks washed and weird, could perhaps change it to something like #cf4529

File: frontend/website/src/components/sideNav.svelte [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+		display: flex;
+		align-items: center;
+		justify-content: center;
+		border-radius: 50%;
+		padding: 0.25rem;
+		font-size: 2rem;
+		width: 3rem;
+		height: 3rem;
+		transition: background-color 0.25s ease, box-shadow 0.25s ease;
+		box-shadow: 0px 6px 20px 2px rgba(255, 115, 87, 0.1);
```
Comment (treuks) [inline frontend/website/src/components/sideNav.svelte:107]: don't think we should have a shadow by default, it looks weird

File: frontend/website/src/components/closeSidebar.svelte (hunk 1 above)
Comment (TroyKomodo) [reply frontend/website/src/components/closeSidebar.svelte:1]: why?

File: frontend/website/src/components/topNav.svelte (hunk 2 above)
Comment (TroyKomodo) [reply frontend/website/src/components/topNav.svelte:111]: hmm, 1px looked a bit small for me
image
image
image
image

File: frontend/website/src/components/closeSidebar.svelte (hunk 1 above)
Comment (TroyKomodo) [reply frontend/website/src/components/closeSidebar.svelte:1]: The advantage of having it 1em means it scales with `font-size:` by an exact factor of the font size.

Comment (treuks) [reply frontend/website/src/components/closeSidebar.svelte:1]: yeah but it just looks straight up bad

Comment (TroyKomodo) [reply frontend/website/src/components/closeSidebar.svelte:1]: test in discord <@!383195095610163200>

Comment (TroyKomodo) [reply frontend/website/src/components/closeSidebar.svelte:1]: im confused how does it look bad? Changing both values to 0.7em means that it scales slightly less than a factor of `font-size` and changing the one value (height) to `0.7em` would make it not a perfect square. By making it a perfect square it just pads the region which is slightly smaller defined in the view box.

Comment (TroyKomodo) [reply frontend/website/src/components/closeSidebar.svelte:1]: if you think the size of the icon should be smaller you should have just said that and we can change it on the side nav and the chat room directly.

Comment (treuks) [reply frontend/website/src/components/closeSidebar.svelte:1]: lol, yeah, sorry, small brain moment

**Review by treuks — APPROVED** [review]
👍

**Pull request merged**